runtime/worker_bootstrap.rs

```diff
@@ -58,28 +57,142 @@ impl Default for BootstrapOptions {
-      "inspectFlag": self.inspect,
-    })
+  pub fn as_v8<'s>(
+    &self,
+    scope: &mut v8::HandleScope<'s>,
+  ) -> v8::Local<'s, v8::Array> {
+    let array = v8::Array::new(scope, 17);
+
+    {
+      let args = v8::Array::new(scope, self.args.len() as i32);
+      for (idx, arg) in self.args.iter().enumerate() {
+        let arg_str = v8::String::new(scope, arg).unwrap();
+        args.set_index(scope, idx as u32, arg_str.into());
+      }
+      array.set_index(scope, 0, args.into());
+    }
+
+    {
+      let val = v8::Integer::new(scope, self.cpu_count as i32);
+      array.set_index(scope, 1, val.into());
+    }
+
+    {
+      let val = v8::Boolean::new(scope, self.debug_flag);
+      array.set_index(scope, 2, val.into());
+    }
+
+    {
+      let val = v8::String::new_from_one_byte(
+        scope,
+        self.runtime_version.as_bytes(),
+        v8::NewStringType::Normal,
+      )
+      .unwrap();
+      array.set_index(scope, 3, val.into());
+    }
+
+    {
+      let val = v8::String::new_from_one_byte(
+        scope,
+        self.locale.as_bytes(),
+        v8::NewStringType::Normal,
+      )
+      .unwrap();
+      array.set_index(scope, 4, val.into());
+    }
+
+    {
+      let val: v8::Local<v8::Value> = if let Some(location) = &self.location {
+        v8::String::new_from_one_byte(
+          scope,
+          location.as_str().as_bytes(),
+          v8::NewStringType::Normal,
+        )
+        .unwrap()
+        .into()
+      } else {
+        v8::undefined(scope).into()
+      };
+
+      array.set_index(scope, 5, val);
+    }
+
+    {
+      let val = v8::Boolean::new(scope, self.no_color);
+      array.set_index(scope, 6, val.into());
+    }
+
+    {
+      let val = v8::Boolean::new(scope, self.is_tty);
+      array.set_index(scope, 7, val.into());
+    }
+
+    {
+      let val = v8::String::new_from_one_byte(
+        scope,
+        self.ts_version.as_bytes(),
+        v8::NewStringType::Normal,
+      )
+      .unwrap();
+      array.set_index(scope, 8, val.into());
+    }
+
+    {
+      let val = v8::Boolean::new(scope, self.unstable);
+      array.set_index(scope, 9, val.into());
+    }
+
+    {
+      let val = v8::Integer::new(scope, std::process::id() as i32);
+      array.set_index(scope, 10, val.into());
+    }
+
+    {
+      let val = v8::Integer::new(scope, ppid() as i32);
+      array.set_index(scope, 11, val.into());
+    }
+
+    {
+      let val = v8::String::new_from_one_byte(
+        scope,
+        env!("TARGET").as_bytes(),
+        v8::NewStringType::Normal,
+      )
+      .unwrap();
+      array.set_index(scope, 12, val.into());
+    }
+
+    {
+      let val = v8::String::new_from_one_byte(
+        scope,
+        deno_core::v8_version().as_bytes(),
+        v8::NewStringType::Normal,
+      )
+      .unwrap();
+      array.set_index(scope, 13, val.into());
+    }
+
+    {
+      let val = v8::String::new_from_one_byte(
+        scope,
+        self.user_agent.as_bytes(),
```

Comment:
This looks like it could be an external static.

Reply:
It can't be static as is, because `self.user_agent` is `String` and not `&'static str`. But we can actually change it to be static str in a follow up.